core/src/main/scala/org/apache/spark/deploy/history/EventLogFileWriters.scala

```diff
@@ -361,7 +362,9 @@ class RollingEventLogFilesWriter(
 
   private def createAppStatusFile(inProgress: Boolean): Unit = {
     val appStatusPath = getAppStatusFilePath(logDirForAppPath, appId, appAttemptId, inProgress)
-    val outputStream = fileSystem.create(appStatusPath)
+    // SPARK-30860: use the class method to avoid the umask causing permission issues
+    val outputStream = FileSystem.create(fileSystem, appStatusPath,
+      EventLogFileWriter.LOG_FILE_PERMISSIONS)
```

Comment:
Actually these permissions are a bit weird for a file; for a file I'd expect 660 (not 770). It's also unnecessary since to delete a file you need write permission to the directory only.

Reply:
Hmm the old `EventLoggingListener` uses 770 also. Maybe it should be changed to 660, and have a separate constant for file permissions and directory permissions.

Reply:
Yeah I just used the existing permissions. Not sure why they were set with the executable bit in the first place. I can change the files to only have 660 if there's no reason not to.

Reply:
Yes please go ahead and fix it altogether.

Reply:
Changed the file permission to 660 and added a separate folder permission for 770